genshin/client/components/geetest/client.py

```diff
@@ -561,3 +570,20 @@ async def login_with_qrcode(self) -> typing.Dict[str, str]:
         }
         self.set_cookies(cookies)
         return cookies
+
+    async def create_geetest(self, cookies: typing.Mapping[str, str]) -> MiyousheGeetest:
+        """Create a geetest challenge."""
+        is_genshin = self.game is Game.GENSHIN
+        headers = {
+            "DS": ds_utility.generate_create_geetest_ds(),
+            "x-rpc-challenge_game": "2" if is_genshin else "6",
+            "x-rpc-page": "v4.1.5-ys_#ys" if is_genshin else "v1.4.1-rpg_#/rpg",
+            "x-rpc-tool-verison": "v4.1.5-ys" if is_genshin else "v1.4.1-rpg",
+            **geetest_utility.CREATE_GEETEST_HEADERS,
+        }
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(routes.CREATE_GEETEST_URL.get_url(), headers=headers, cookies=cookies) as r:
```

Comment:
What is the purpose of using `cookies` here? Is it required for hoyo API to identify the user?

Reply:
yes

Reply:
K, but why don't we use cookies from the client rather then passing them directly?

Reply:
cuz idk how to access the client's cookies